spread/virus3/Cell.java

```diff
@@ -0,0 +1,65 @@
+package sim.app.virus3;
+
+import sim.util.*;
+import sim.engine.*;
+import java.awt.*;
+import sim.portrayal.*;
+
+// Based on Human
+public class Cell extends BioAgent
+    {
+    private static final long serialVersionUID = 1;
+    
+    // If the state is true the cell is infected
+    protected boolean infected = false;
+    // This method checks the status of the cell
+    public final boolean isInfected() { return infected; }
+    // This method may be used to change the status of the cell
+    public final void setInfected( boolean b ) { infected = b; }
+    
+    // Main class
+    public Cell( SimpleVirusDemo agent, String id, Double2D location ) 
+        {
+        super( agent, id, location );
+        try
+            {
+            // This returns an integer starting at the fifth location in the string id
+            // Changed 5 to 4 (length of word Human vs Cell)
+            // The id is set in SimpleVirusDemo when a new agent is made
+            intID = Integer.parseInt( id.substring(4) ); // "Cell"
+            }
+        catch( Exception e )
+            {
+            throw new RuntimeException(e);
+            }
+        }
+    
+    public void step( final SimState state )
```

Comment:
Oh, I was just looking at the code that was shown in the email...I'll get rid of the empty braces in future versions